src/Symfony/Component/Mailer/EventListener/EnvelopeListener.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Mailer\Event\MessageEvent;
+use Symfony\Component\Mime\Address;
+
+/**
+ * Manipulates the Envelope of a Message.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @experimental in 4.3
+ */
+class EnvelopeListener implements EventSubscriberInterface
```

Comment:
Just a question, why not EnvelopeSubscriber in namespace EventSubscriber?

Reply:
Because the class is a listener with the possibility to use it as a subscriber.